jenkins/common.groovy

```diff
@@ -131,6 +131,39 @@ def setup() {
   }
 }
 
+def checkForBrokenLinks(String buildLogPath, String envName) {
+  // Look for broken links
+  def csvFileName = "${envName}-broken-links.csv" // For use within the docker container
+  def csvFile = "${WORKSPACE}/vets-website/${csvFileName}" // For use outside of the docker context
+
+  // Ensure the file isn't there if we had to rebuild
+  if (fileExists(csvFile)) {
+    sh "rm /application/${csvFileName}"
+  }
+
+  // Output a csv file with the broken links
+  sh "cd /application && jenkins/glean-broken-links.sh ${buildLogPath} ${csvFileName}"
+  if (fileExists(csvFile)) {
+    echo "Found broken links; notifying the Slack channel."
+    // TODO: Move this slackUploadFile to cacheDrupalContent and update the echo statement above
+    // slackUploadFile(filePath: csvFile, channel: 'dev_null', failOnError: true, initialComment: "Found broken links in the ${envName} build on `${env.BRANCH_NAME}`.")
+
+    // Until slackUploadFile works...
+    def linkCount = sh(returnStdout: true, script: "cd /application && wc -l ${csvFileName} | cut -d ' ' -f1") as Integer
+    slackSend message: "@here ${linkCount - 1} broken links found in the ${envName} build on `${env.BRANCH_NAME}`\n${env.RUN_DISPLAY_URL}".stripMargin(),
+      color: 'danger',
+      failOnError: true,
+      channel: 'cms-general'
+
+    // Only break the build if broken links are found in master
+    if (isDeployable()) {
+      throw new Exception('Broken links found')
+    }
```

Comment:
If I'm following correctly, this means it breaks due to broken links only on `vagovprod` builds and only on the master branch. What if a feature branch is responsible for the broken links due to a template error, GraphQL query errors, or something else? Won't that work make its way into master?

Reply:
Unfortunately yes. :disappointed:

The _hope_ is that the alert to the #cms-general channel would be enough to catch that, but it's definitely imperfect.

Can you think of a way to catch broken links caused by a `vets-website` issue and not by the CMS content?

Reply:
My concern is just that template-changing pull requests will no longer be forced to fix broken links if they have them, and no one will notice until time for the daily deployment

Reply:
Totally agree. The alternative is having content editors break every team's `vets-website` builds, though.